qutebrowser/browser/webengine/interceptor.py

```diff
@@ -29,8 +31,34 @@
 from qutebrowser.extensions import interceptors
 
 
-class RequestInterceptor(QWebEngineUrlRequestInterceptor):
+@attr.s
+class WebEngineRequest(interceptors.Request):
+    """Overrides for QtWebEngine."""
+
+    WHITELIST_REQUEST_METHODS = {QByteArray(b'GET'), QByteArray(b'HEAD')}
+
+    #: If this request has been redirected already
+    _webengine_info = attr.ib(default=None, type=QWebEngineUrlRequestInfo)
```

Comment:
Why have a default at all if you're checking whether it's been set by hand anyways?

Reply:
attrs yells at me about non-default args not being after default ones, it dosen't seem smart enough to properly handle this case. I don't feel like restructuring all of this because attrs is bad though.

```
Traceback (most recent call last):
  File "/usr/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/jay/Code/qutebrowser/qutebrowser/__main__.py", line 29, in <module>
    sys.exit(qutebrowser.qutebrowser.main())
  File "/home/jay/Code/qutebrowser/qutebrowser/qutebrowser.py", line 194, in main
    return app.run(args)
  File "/home/jay/Code/qutebrowser/qutebrowser/app.py", line 143, in run
    init(args, crash_handler)
  File "/home/jay/Code/qutebrowser/qutebrowser/app.py", line 171, in init
    _init_modules(args, crash_handler)
  File "/home/jay/Code/qutebrowser/qutebrowser/app.py", line 500, in _init_modules
    browsertab.init()
  File "/home/jay/Code/qutebrowser/qutebrowser/browser/browsertab.py", line 82, in init
    from qutebrowser.browser.webengine import webenginetab
  File "/home/jay/Code/qutebrowser/qutebrowser/browser/webengine/webenginetab.py", line 36, in <module>
    from qutebrowser.browser.webengine import (webview, webengineelem, tabhistory,
  File "/home/jay/Code/qutebrowser/qutebrowser/browser/webengine/interceptor.py", line 35, in <module>
    class WebEngineRequest(interceptors.Request):
  File "/usr/lib/python3/dist-packages/attr/_make.py", line 904, in attrs
    return wrap(maybe_cls)
  File "/usr/lib/python3/dist-packages/attr/_make.py", line 855, in wrap
    cache_hash,
  File "/usr/lib/python3/dist-packages/attr/_make.py", line 471, in __init__
    cls, these, auto_attribs, kw_only
  File "/usr/lib/python3/dist-packages/attr/_make.py", line 400, in _transform_attrs
    "default value or factory.  Attribute in question: %r" % (a,)
ValueError: No mandatory attributes allowed after an attribute with a default value or factory.  Attribute in question: Attribute(name='_webengine_info', default=NOTHING, validator=None, repr=True, cmp=True, hash=None, init=True, metadata=mappingproxy({}), type=<class 'PyQt5.QtWebEngineCore.QWebEngineUrlRequestInfo'>, converter=None, kw_only=False)
zsh: segmentation fault  python3 -m qutebrowser -T
```